zerver/tests/test_realm_linkifiers.py

```diff
@@ -114,3 +116,55 @@ def test_delete(self) -> None:
         result = self.client_delete(f"/json/realm/filters/{linkifier_id}")
         self.assert_json_success(result)
         self.assertEqual(RealmFilter.objects.count(), linkifiers_count - 1)
+
+    def test_update(self) -> None:
+        self.login("iago")
+        realm = get_realm("zulip")
+        linkifier_id = do_add_linkifier(
+            realm, "#(?P<id>[123])", "https://realm.com/my_realm_filter/pull/%(id)s"
+        )
+        result = self.client_get("/json/realm/linkifiers")
+        self.assert_json_success(result)
+
+        do_update_linkifier(
+            realm,
+            linkifier_id,
+            "#(?P<id>[0-9]+)",
+            "https://realm.com/my_realm_filter/issues/%(id)s",
+        )
+        result = self.client_get("/json/realm/linkifiers")
+        self.assert_json_success(result)
+        # Verify that the linkifier is updated accordingly.
+        linkifier = result.json()["linkifiers"]
+        self.assertEqual(len(linkifier), 1)
+        self.assertEqual(linkifier[0]["pattern"], "#(?P<id>[0-9]+)")
+        self.assertEqual(
+            linkifier[0]["url_format"], "https://realm.com/my_realm_filter/issues/%(id)s"
+        )
+
+        data = {
+            "pattern": r"ZUL-(?P<id>\d++)",
+            "url_format_string": "https://realm.com/my_realm_filter/%(id)s",
+        }
+        result = self.client_patch(f"/json/realm/filters/{linkifier_id}", info=data)
+        self.assert_json_error(
+            result, "Invalid linkifier pattern.  Valid characters are [ a-zA-Z_#=/:+!-]."
```

Comment:
These validations are done since a new API is added (hence I've used `client_patch`).
All the validations/error validations are done to verify that they work correctly for the new added API.
I've not repeated any validation like `create` tests; here, I've just made sure that we do all types of validations (via this new API).